unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderEntity/DecentralandEntityToEdit.cs

```diff
@@ -0,0 +1,295 @@
+using DCL;
+using DCL.Models;
+using DCL.Components;
+using DCL.Controllers;
+using DCL.Helpers;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class DecentralandEntityToEdit : EditableEntity
+{
+    public string entityUniqueId;
+
+    public System.Action<DecentralandEntityToEdit> onStatusUpdate,OnDelete;
+
+    private bool isLocked = false;
+    public bool IsLocked
+    {
+        get
+        {
+            return isLocked;
+        }
+        set
+        {
+            isLocked = value;
+            onStatusUpdate?.Invoke(this);
+        }
+    }
+
+    private bool isSelected = false;
+    public bool IsSelected
+    {
+        get
+        {
+            return isSelected;
+        }
+        set
+        {
+            isSelected = value;
+            onStatusUpdate?.Invoke(this);
+        }
+    }
+
+    private bool isNew = false;
+    public bool IsNew
+    {
+        get
+        {
+            return isNew;
+        }
+        set
+        {
+            isNew = value;
+            onStatusUpdate?.Invoke(this);
+        }
+    }
+
+    private bool isVisible = true;
+    public bool IsVisible
+    {
+        get
+        {
+            return isVisible;
+        }
+        set
+        {
+            isVisible = value;
+            onStatusUpdate?.Invoke(this);
+        }
+    }
+
+
+    private bool isVoxel = false;
+    public bool IsVoxel
+    {
+        get
+        {
+            return isVoxel;
+        }
+        set
+        {
+            isVoxel = value;
+        }
+    }
+
+    Transform originalParent;
+
+    Material[] originalMaterials;
+
+    Material editMaterial;
+
+    Dictionary<string, GameObject> collidersDictionary = new Dictionary<string, GameObject>();
+
+    public void Init(DecentralandEntity _entity, Material _editMaterial)
```

Comment:
I don't understand what you mean there, If I understand it ( I may be wrong), you will have two instances of `DecentralandEntityToEdit `referencing the same `DecentralandEntity`, if you are worried about the dictionary, it is for the children since each ID is unique you won't have children with the same ID of the parent. Let me know if this answer it